### DISCUSSION TITLE
Self-destroy pipeline after n time or destroy multiple pipelines in one command

### DISCUSSION THREAD

**Discussion by code2fire00** [concourse/concourse#8402] [Help & Support] [answered]
Hi pilots, I'm looking for a way to spend less time destroying pipelines manually.

Is there a way to configure a pipeline so it automatically destroys itself given n period of time?

If not, that brings me to my second question:
Given multiple pipelines that begins with the same prefix ex: my-pipeline-one, my-pipeline-two ... 
is there a command to destroy them at once ?

Thank you very much for taking your time to answer. :)

**Comment by marco-m** [comment, marked as answer]
Hello,

> Is there a way to configure a pipeline so it automatically destroys itself given n period of time?

I don't think so. The following issues are similar: https://github.com/concourse/concourse/issues/3281 and https://github.com/concourse/concourse/issues/4578.

> Given multiple pipelines that begins with the same prefix ex: my-pipeline-one, my-pipeline-two ...
> is there a command to destroy them at once ?

No, but it is easy to write a shell script.

Say we have some pipelines:

```
$ fly -t main pipelines
id  name   paused  public  last updated
4   foo-1  no     no      2022-06-07 18:13:13 +0200 CEST
5   foo-2  no     no      2022-06-07 18:13:22 +0200 CEST
6   bar    no     no      2022-06-07 18:13:29 +0200 CEST
```

And we want to destroy all with prefix `foo`. This will filter the ones we want:

```
$ fly -t main pipelines | awk '{print $2}' | grep foo
foo-1
foo-2
```

We can pass the same to a loop that invokes destroy-pipeline:

```
$ for p in (fly -t main pipelines | awk '{print $2}' | grep foo);
    echo destroying $p;
    fly -t main destroy-pipeline -p $p;
end
```

The above will ask for confirmation each time. If you are _sure_ of your filter, you can also pass `--non-interactive` to the command.

The shell in the example is fish; to use a bash-like the syntax is slightly different but the logic is the same.

**Reply by code2fire00** [reply]
Thanks for your answer!
I really like your idea of the optional pipeline TTL, wouldve been awesome to have something like that.
